sklearn/cluster/tests/test_hierarchical.py

```diff
@@ -598,3 +598,15 @@ def increment(self, *args, **kwargs):
     linkage_tree(X, connectivity=connectivity, affinity=fa.increment)
 
     assert_equal(fa.counter, 3)
+
+
+def test_n_components_deprecation():
+    # Test that a Deprecation warning is thrown when n_components_
+    # attribute is accessed
+
+    X = np.array([[1, 2], [1, 4], [1, 0], [4, 2]])
+    agc = AgglomerativeClustering().fit(X)
+
+    with pytest.warns(DeprecationWarning):
```

Comment:
Please check the warning message as well with the `match` parameter (see [here](https://docs.pytest.org/en/latest/warnings.html#warns))

Reply:
Again thanks Nicolas, I had not considered that case - this was addressed in commit [#14ac792](https://github.com/scikit-learn/scikit-learn/pull/13427/commits/14ac792131ba8f6424eb42f912070733db423c9b)